src/libcore/option.rs

```diff
@@ -913,6 +913,65 @@ impl<T> Option<T> {
     pub fn replace(&mut self, value: T) -> Option<T> {
         mem::replace(self, Some(value))
     }
+
+    /// Zips `self` with another `Option`.
+    ///
+    /// Returns `Some((_, _))` when both `self` and `other`
+    /// are `Some(_)`, otherwise return `None`.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// #![feature(option_zip)]
+    /// let x = Some(1);
+    /// let y = Some("hi");
+    /// let z = None::<u8>;
+    ///
+    /// assert_eq!(x.zip(y), Some((1, "hi")));
+    /// assert_eq!(x.zip(z), None);
+    /// ```
+    #[inline]
+    #[unstable(feature = "option_zip", issue = "none")]
+    pub fn zip<U>(self, other: Option<U>) -> Option<(T, U)> {
+        self.zip_with(other, |a, b| (a, b))
+    }
+
+    /// Zips `self` and another `Option` with function `f`.
+    ///
+    /// Returns `Some(_)` when both `self` and `other`
+    /// are `Some(_)`, otherwise return `None`.
```

Comment:
This explanation is not quite sufficient I think. A small improvement would be "Returns `Some(f(_, _))`. But why not more explicit, like:

> If `self` is `Some(s)` and `other` is `Some(o)`, this method returns `Some(f(s, o))`. Otherwise, `None` is returned.

If you only change it slightly, the same "when" -> "if" and "returnS" comment as above apply here.

Reply:
I've also changed the docs of `zip` for docs to be more consistent